spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/server/MissingWebServerFactoryBeanException.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2012-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.web.server;
+
+import org.springframework.beans.FatalBeanException;
+import org.springframework.boot.WebApplicationType;
+
+/**
+ * Throws exception when web server factory bean is missing.
+ *
+ * @author Guirong Hu
+ * @since 2.x
+ */
+@SuppressWarnings("serial")
+public class MissingWebServerFactoryBeanException extends FatalBeanException {
+
+	private final WebApplicationType webApplicationType;
+
+	/**
+	 * Create a new {@code MissingWebServerFactoryBeanException} with the specified detail
+	 * message and no root cause and the given type of web application.
+	 * @param msg the detail message
+	 * @param webApplicationType the web application type
+	 */
+	public MissingWebServerFactoryBeanException(String msg, WebApplicationType webApplicationType) {
+		this(msg, null, webApplicationType);
+	}
+
+	/**
+	 * Create a new {@code MissingWebServerFactoryBeanException} with the specified detail
+	 * message and the given root cause and the given type of web application.
+	 * @param msg the detail message
+	 * @param cause the root cause
+	 * @param webApplicationType the web application type
+	 */
+	public MissingWebServerFactoryBeanException(String msg, Throwable cause, WebApplicationType webApplicationType) {
```

Comment:
This exception will never have a cause as currently used so I think we could remove this constructor for now at least.

Reply:
The same as above.